src/demoAskScriptServer.ts

```diff
@@ -0,0 +1,152 @@
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import express from 'express';
+
+import { askCodeToSource } from './askcode';
+import {
+  resources as builtInResources,
+  runUntyped,
+  resource,
+  Resources,
+} from './askvm';
+import { parse as parseAskScript, AskScriptCode } from './askscript';
+
+import chalk = require('chalk');
+import { customAlphabet } from 'nanoid';
+
+const nanoid = customAlphabet('1234567890abcdef', 8);
+
+const values = {
+  clientNames: ['a', 'b', 'c', 'd', 'e', 'f', 'g'],
+  hello: 'Hi! This is a AskVM server running on localhost',
+  revPerClient: {
+    a: 426,
+    b: 35,
+    c: 446,
+    d: 246,
+    e: 133,
+    f: 136,
+    g: 53,
+  },
+
+  firstName: 'Luke',
+  middleName: 'LukeLuke',
+  lastName: 'Skywalker',
+  fullName: 'Luke Skywalker',
+  parents: [
+    {
+      firstName: 'Padmé',
+      lastName: 'Amidala',
+    },
+    {
+      firstName: 'Anakin',
+      lastName: 'Skywalker',
+    },
+  ],
+  friends: [
+    {
+      id: 0,
+      firstName: 'Padmé',
+      lastName: 'Amidala',
+    },
+    {
+      id: 1,
+      firstName: 'Anakin',
+      lastName: 'Skywalker',
+    },
+  ],
+};
+
+export const customResources: Resources = {
+  withSmile: resource<string, [string]>({
+    resolver: async (s: string): Promise<string> => {
+      return ':) ' + s + ' :)';
+    },
+  }),
+
+  hi: resource<string, []>({
+    resolver: async (): Promise<string> => {
+      return "Hi, this is AskQL server! It's " + new Date().toString();
+    },
+  }),
+};
+
+const baseEnvironment = {
+  resources: { ...builtInResources, ...customResources },
+  values,
+};
+
+const app = express();
+app.use(cors());
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+
+app.get('/', async (req, res) => {
+  res.type('html');
+  res.send(
+    '<html><head><title>AskQL</title></head><body><p><b>Ask</b> me anything!</p><p><a href="http://askql.org/">AskQL website</a></p></body></html>'
+  );
+});
+
+app.post('/askscript', async (req, res) => {
```

Comment:
Since we're probably going to support different options for providing the query, let's maybe use a more generic endpoint name - I suggest `/ask`

Reply:
> > In which milestone would you like to deploy it to Netlify?
> 
> I used Netlify as an example. I assume this playground backend is going to be deployed to a server. I am wondering how are we going to get these logs from that server.
> 
> @czerwinskilukasz1 Łukasz, here's an example solution for logging [http://github.com/winstonjs/winston](https://github.com/winstonjs/winston), there are more

Great, let's discuss it in a separate issue. https://github.com/xFAANG/askql/issues/215

Reply:
> Please use @YonatanKra's change

@mhagmajer , I think Yonatan's server is not ready to be used in AskScript Playground. It assumes the code is in AskCode and only if parsing it fails, it tries to parse it as an AskScript program. Therefore, it cannot be used as a server for AskScript Playground, where user is expecting to write only valid AskScript programs.

Example:
Typing in the AskScript Playground:
`ask(let('plus',fun(let('a',get('$0')),let('b',get('$1')),let('c',get('$2')),return(call(get('sum'),call(get('sum'),get('a'),get('b')),get('c'))))),call(get('plus'),2,3.6,4))` should result in an error but if I used Yonatan's server, user would see result: `9.6`

Reply:
Let's keep it /askscript for now, because it signals that Ask*Script* is accepted not AskCode. I had created an issue for sending AskCode to the server instead of AskScript. We could change the name of the endpoint then (or introduce /ask in addition to /askscript), let's keep the conversation here: https://github.com/xFAANG/askql/issues/187